frontend/packages/console-app/src/components/volume-snapshot/create-volume-snapshot/create-volume-snapshot.tsx

```diff
@@ -0,0 +1,315 @@
+  };
+
+  return (
+    <ListDropdown
+      {...props}
+      desc="Volume Snapshot Class with same provisioner as claim"
+      dataFilter={filter}
+      resources={resources}
+      selectedKeyKind={kind}
+      placeholder="Select snapshot class"
+      selectedKey={selectedKey}
+    />
+  );
+};
+
+const PVCSummary: React.FC<PVCSummaryProps> = ({ persistentVolumeClaim }) => {
+  const storageClass = persistentVolumeClaim?.spec?.storageClassName;
+  const requestedCapacity = persistentVolumeClaim?.spec?.resources?.requests?.storage;
+  const accessModes =
+    persistentVolumeClaim?.spec?.accessModes.reduce(
+      (allModes: string, mode: string) => `${mode} ${allModes}`,
+      '',
+    ) || '-';
+  const volumeMode = persistentVolumeClaim?.spec?.volumeMode;
+  return (
+    <dl>
+      <dt className="co-volume-snapshot__details-body">Persistent Volume Claim Details</dt>
+      <dt>Name</dt>
+      <dd>
+        <ResourceIcon kind={PersistentVolumeClaimModel.kind} />
+        {getName(persistentVolumeClaim)}
+      </dd>
+      <dt>Namespace</dt>
+      <dd>
+        <ResourceIcon kind={NamespaceModel.kind} />
+        {getNamespace(persistentVolumeClaim)}
+      </dd>
+      <dt>Status</dt>
+      <dd>
+        <PVCStatus pvc={persistentVolumeClaim} />
+      </dd>
+      <dt>Storage Class</dt>
+      <dd>
+        <ResourceIcon kind={StorageClassModel.kind} />
+        {storageClass}
+      </dd>
+      <dt>Requested Capacity</dt>
+      <dd>{requestedCapacity}</dd>
+      <dt>Access Mode</dt>
+      <dd>{accessModes}</dd>
+      <dt>Volume Mode</dt>
+      <dd>{volumeMode}</dd>
+    </dl>
+  );
+};
+
+const CreateSnapshotForm = withHandlePromise((props: SnapshotResourceProps) => {
+  const { resourceName, namespace, kindObj, handlePromise, inProgress, errorMessage } = props;
+
+  const [pvcName, setPVCName] = React.useState(resourceName);
+  const [pvcObj, setPVCObj] = React.useState<PersistentVolumeClaimKind>(null);
+  const [snapshotName, setSnapshotName] = React.useState(`${pvcName || 'pvc'}-snapshot`);
+  const [snapshotClassName, setSnapshotClassName] = React.useState<string>();
+  const [pvcListObj, setListPVCObj] = React.useState(null);
+  const title = 'Create VolumeSnapshot';
+
+  React.useEffect(() => {
+    if (pvcName && namespace) {
+      k8sGet(PersistentVolumeClaimModel, pvcName, namespace)
+        .then((pvc) => setPVCObj(pvc))
+        .catch((error) =>
+          // eslint-disable-next-line no-console
+          console.error(error),
+        );
+    } else {
+      k8sGet(PersistentVolumeClaimModel, undefined, undefined)
+        .then((list) => setListPVCObj(list))
+        .catch((error) =>
+          // eslint-disable-next-line no-console
+          console.error(error),
+        );
+    }
+  }, [pvcName, namespace]);
+
+  const handleSnapshotName: React.ReactEventHandler<HTMLInputElement> = (event) =>
+    setSnapshotName(event.currentTarget.value);
+
+  const handlePVCName = (obj: string) => {
```

Comment:
```suggestion
  const handlePVCName = (name: string) => {
```
I would expect it to be `object` if the parameter name is `obj`.